src/collection.rs

```diff
@@ -204,10 +280,40 @@ impl<'a, C: ClientExt> Collection<'a, C> {
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn checksum(&self) {
-        unimplemented!()
+    pub async fn checksum(&self) -> Result<CollectionChecksum, ClientError> {
+        self.checksum_with_options(None, None).await
     }
 
+    // By setting the optional query parameter withRevisions to true, then revision
+    // ids (_rev system attributes) are included in the checksumming.
+    //
+    // By providing the optional query parameter withData with a value of true, the
+    // user-defined document attributes will be included in the calculation too.
+    // Note: Including user-defined attributes will make the checksumming slower.
+    #[maybe_async]
+    pub async fn checksum_with_options(
+        &self,
+        with_revisions: Option<bool>,
```

Comment:
Why wrap the bool in `Option`? Can we use bool only?

Reply:
Hey, I was thinking it was "optional" parameters, so I used options. But you are right, we could use only bool. I'll make the try with only bool and see how it looks like.